### PR TITLE
Category addition on newroom pages and also in feed block

### LINKED ISSUE #361: Add support for Categories for news articles
Display the category if defined on the News cards.
image

Add filter on categories
image

Add support for NewsCategory in metadata like this:
image

### PR DESCRIPTION
**Pull request by jindaliiita**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after), along with a short summary of changes:

## Issue

Fixes #361 

## Changelog:
_Please enter each change as a new bullet point_

## Test URLs:
- Original: https://www.sunstar.com/newsroom/news/20150430
- Before: https://main--sunstar--hlxsites.hlx.live/_drafts/piyush/20150430
- After: https://category-changes--sunstar--hlxsites.hlx.live/_drafts/piyush/20150430

## Library

- [ ] New Blocks introduced in this PR
      If yes, please provide details below
**Block Name** : For e.g. _cards_
- [ ] Documented in sidekick library

- [ ] New variations introduced in this PR
**Variation Name** :  For e.g. _cards (grid)_
- [ ] Documented in sidekick library

- [ ] New mixins introduced in this PR
**Mixin Name** :  For e.g. _compact, white_
- [ ] Documented in sidekick library

### CODE REVIEW

**Comment by jindaliiita** [comment]
Test failures can be ignored as of now.

File: blocks/social/social.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default async function decorate(block) {
       .filter(Boolean)
       .join('-');
 
-    if (firstH1) {
+    if (firstH1 && type.toLowerCase() !== 'newsroom') {
```
[the remark this answers is not in the document]
Comment (jindaliiita) [reply blocks/social/social.js:108]: On healthy thinking pages we show category as anchor tags. And these categories have a page associated with them
But for news articles we don't have category pages. Thats why i have made this check
@ehrro Can you please check

**Review by sdmcraft — COMMENTED** [review]
Couple of open comments but overall looks good..

**Review by sdmcraft — APPROVED** [review]

File: blocks/feed/feed.js [hunk 2]
```diff
@@ -2,37 +2,53 @@ import {
...
 
 // Result parsers parse the query results into a format that can be used by the block builder for
 // the specific block types
 const resultParsers = {
   // Parse results into a cards block
-  cards: (results, blockCfg, variation = '') => {
+  cards: async (results, blockCfg, variation = '') => {
     const blockContents = [];
-    results.forEach((result) => {
+    for (let index = 0; index < results.length; index += 1) {
```
[the remark this answers is not in the document]
Comment (jindaliiita) [reply blocks/feed/feed.js:14]: Implemented in the same way but somehow its not working as expected

File: blocks/feed/feed.js [hunk 3]
```diff
@@ -2,37 +2,53 @@ import {
...
-            div.classList.add('title');
-            div.textContent = result[fieldName];
+          // eslint-disable-next-line no-lonely-if
+          if (fieldName === 'category') {
+            // Field name category handling
+            if (result[fieldName] !== '0') {
+              const anchor = document.createElement('a');
+              anchor.classList.add('category');
+              // eslint-disable-next-line no-await-in-loop
+              const categories = await fetchTagsOrCategories([result[fieldName]], 'categories', '', getLanguage());
```
[the remark this answers is not in the document]
Comment (jindaliiita) [reply blocks/feed/feed.js:35]: Tracked via https://github.com/hlxsites/sunstar/issues/478

**Pull request merged**